lib/improver/wind_direction.py

```diff
@@ -0,0 +1,404 @@
+        x_coord_name = cube_ens_wdir.coord(axis="x").name()
+        for slice_ens_wdir in cube_ens_wdir.slices(["realization",
+                                                    y_coord_name,
+                                                    x_coord_name]):
+            # Extract wind direction data.
+            wind_dir_deg = slice_ens_wdir.data
+            realization_axis = slice_ens_wdir.coord_dims('realization')[0]
+
+            # Copies input cube and remove realization dimension to create
+            # cubes for storing results.
+            slice_mean_wdir = next(slice_ens_wdir.slices_over("realization"))
+            slice_mean_wdir.remove_coord('realization')
+            slice_r_vals = slice_mean_wdir.copy()
+            slice_std_dev = slice_mean_wdir.copy()
+
+            # Convert wind direction from degrees to complex numbers.
+            wind_dir_complex = WindDirection.deg_to_complex(wind_dir_deg)
+
+            # Find the complex average -  which actually signifies a point
+            # between all of the data points in POLAR co-cordinates.
+            # NOT the average DEGREE ANGLE.
+            wind_dir_complex_mean = np.mean(wind_dir_complex,
+                                            axis=realization_axis)
+
+            # Convert complex average values to degrees to produce average
+            # wind direction.
+            wind_dir_deg_mean = WindDirection.complex_to_deg(
+                wind_dir_complex_mean)
+
+            # Find radius values for wind direction average.
+            r_vals = WindDirection.find_r_values(wind_dir_complex_mean)
+
+            # Calculate standard deviation from polar mean.
+            # ToDo: This will still need some further investigation.
+            #        This is will be the subject of another ticket.
+            polar_mean_std_dev = WindDirection.calc_polar_mean_std_dev(
+                wind_dir_complex, wind_dir_deg_mean, r_vals, r_thresh,
+                realization_axis)
+
+            # Finds any meaningless averages and subistuite with
+            # the wind direction taken from the first ensemble member.
+            wind_dir_deg_mean = WindDirection.wind_dir_decider(
+                wind_dir_deg, wind_dir_deg_mean, r_vals, r_thresh)
+
+            # Save data into cubes.
+            slice_mean_wdir.data = wind_dir_deg_mean
+            slice_r_vals.data = r_vals
+            slice_std_dev.data = polar_mean_std_dev
+            # Append to cubelists.
+            wdir_cube_list.append(slice_mean_wdir)
+            r_vals_cube_list.append(slice_r_vals)
+            std_dev_cube_list.append(slice_std_dev)
+
+        # Combine cubelists into cube and outputs.
+        cube_mean_wdir = wdir_cube_list.merge_cube()
+        cube_r_vals = r_vals_cube_list.merge_cube()
+        cube_polar_mean_std_dev = std_dev_cube_list.merge_cube()
+
+        # Change cube identifiers.
+        cube_mean_wdir.long_name = "Average wind direction"
```

Comment:
I'm slightly concerned about these metadata changes. I think we should keep the standard name for the wind direction cube (especially as this should be `wind_from_direction` or `wind_to_direction`). We can always put in a cell method to say it is a mean. We should also set the units to degrees (we may need to convert the input to degrees as well?).  I'm not so concerned about the other cubes that are output as I think they are only intermediate results, and probably won't be saved into netCDF files. @carolinejjones  any opinions?

Reply:
I'll keep the standard name and put in a cell method to say its mean.
As for your comment on setting units to degrees. The input is already in degrees, I created the output cube by copying the input cube and removing the realization dimension, so the output cube units are degrees.
You're right that the other output cubes are only intermediate results, they will be investigated further as part of a separate ticket and then we'll decide what to do with that data then.

Reply:
If you have a cube that is not in degrees to start with then you end up with the wrong units in your output cube. Maybe you should just check that your input cube in in degree. You could do something like:
`try:
     cube.convert_units("degrees")
except some kind of error
    raise warning "Input cube cannot be converted to degrees"
`
Just to be sure. What do you think?